### PR TITLE
fix(deno): do not set vite.ssr.noExternal

### LINKED ISSUE #8390: Astro 3 - build failed with Deno adapter and React integration
### Astro Info

```block
Astro                    v3.0.7
Node                     v20.5.1
System                   Windows (x64)
Package Manager          pnpm
Output                   server
Adapter                  @astrojs/deno
Integrations             @astrojs/react
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

Error when run pnpm build

```
 error   Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. Received protocol 'astro:'
Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. Received protocol 'astro:'
    at new NodeError (node:internal/errors:405:5)
    at throwIfUnsupportedURLScheme (node:internal/modules/esm/load:138:11)
    at defaultLoad (node:internal/modules/esm/load:89:3)
    at DefaultModuleLoader.load (node:internal/modules/esm/loader:263:26)
    at DefaultModuleLoader.moduleProvider (node:internal/modules/esm/loader:179:22)
    at new ModuleJob (node:internal/modules/esm/module_job:63:26)
    at #createModuleJob (node:internal/modules/esm/loader:203:17)
    at DefaultModuleLoader.getJobFromResolveResult (node:internal/modules/esm/loader:156:34)
    at DefaultModuleLoader.getModuleJob (node:internal/modules/esm/loader:141:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:33)
 ELIFECYCLE  Command failed with exit code 1.
```

### What's the expected result?

Able to integrate React with Deno adapter

### Link to Minimal Reproducible Example

https://stackblitz.com/github/flamrdevs/astro-deno-react

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes
- Fixes #8390

## Testing
Could use more tests, not sure if this is the time for it.

## Docs
Does not affect usage.

### CODE REVIEW

File: packages/integrations/deno/src/index.ts [hunk 1]
```diff
@@ -149,9 +150,6 @@ export default function createIntegration(args?: Options): AstroIntegration {
 							(vite.resolve.alias as Record<string, string>)[alias.find] = alias.replacement;
 						}
 					}
-					vite.ssr = {
-						noExternal: COMPATIBLE_NODE_MODULES,
-					};
```
Comment (lilnasy) [inline packages/integrations/deno/src/index.ts:154]: `noExternal` was set to true originally. I had to change it to the list of bare node built-in modules in #7288. But apparently, it doesn't need to be set at all.

**Comment by ematipico** [comment]
Are we sure about this fix? The adapter is about to be moved https://github.com/withastro/astro/pull/8559

**Comment by lilnasy** [comment]
I know but unfortunately, this is a build-breaking bug.

**Comment by bluwy** [comment]
The changes looks fine to me. I think the previous implementation _probably_ wasn't affecting much either, but I don't quite understand how this PR fixes #8390. How was `noExternal`-ing the node modules causing `astro:*` imports to be left in the built bundle before?

**Review by ematipico — APPROVED** [review]
Checked with @matthewp on Discord and we are good to go with merging this PR. Let's 🚢 i!

**Pull request merged**